hoomd/hpmc/ComputeSDF.h

```diff
@@ -364,13 +406,180 @@ template<class Shape> void ComputeSDF<Shape>::countHistogram(uint64_t timestep)
                     }
                 } // end loop over AABB nodes
             }     // end loop over images
-
-        // record the minimum bin
-        if ((unsigned int)min_bin < m_hist.size())
+        if (min_bin < m_hist.size())
+            {
             m_hist[min_bin]++;
+            }
+        } // end loop over all particles
+    }     // end countHistogramBinarySearch()
+
+template<class Shape> void ComputeSDF<Shape>::countHistogramLinearSearch(uint64_t timestep)
+    {
+    // update the aabb tree
+    const hoomd::detail::AABBTree& aabb_tree = m_mc->buildAABBTree();
+
+    // update the image list
+    const std::vector<vec3<Scalar>>& image_list = m_mc->updateImageList();
+
+    Scalar extra_width = m_xmax / (1 - m_xmax) * this->getMaxInteractionDiameter();
```

Comment:
We could save a few cycles by making `max_diam` a member variable (`m_max_diam`) and reusing the computation here.